### PR TITLE
destroy dropdown panel on close

### PR DESCRIPTION
**Pull request by anjmao**
With this PR dropdown panel will be added to the DOM only when dropdown is opened.

### CODE REVIEW

**Comment by anjmao** [comment]
1) https://github.com/ng-select/ng-select/issues/284

File: src/ng-select/dropdown-panel.component.ts [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    }
+
+    ngOnInit() {
+        this._handleScroll();
+        if (this.appendTo) {
+            this._handleAppendTo();
+        }
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
```
Comment (varnastadeus) [inline src/ng-select/dropdown-panel.component.ts]: this method isn't tested

File: src/ng-select/dropdown-panel.component.ts [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    }
+
+    ngOnDestroy() {
+        this._disposeDocumentResizeListener();
+        this._disposeScrollListener();
+        this._elementRef.nativeElement.remove();
+    }
+
+    refresh() {
+        if (IS_TEST) {
```
Comment (varnastadeus) [inline src/ng-select/dropdown-panel.component.ts]: can anything else can be done here?

File: src/ng-select/ng-dropdown-panel.component.ts [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+} from '@angular/core';
+
+import { VirtualScrollService } from './virtual-scroll.service';
+import { NgOption } from './ng-select.types';
+import { NgSelectComponent, DropdownPosition } from './ng-select.component';
+import { ItemsList } from './items-list';
+import { WindowService } from './window.service';
+
+@Component({
+    providers: [VirtualScrollService],
```
Comment (varnastadeus) [inline src/ng-select/ng-dropdown-panel.component.ts]: Here each component instance will have separate instance of `VirtualScrollService`, not sure if this is a good idea

File: src/ng-select/dropdown-panel.component.ts [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+import { VirtualScrollService } from './virtual-scroll.service';
+import { NgOption } from './ng-select.types';
+import { NgSelectComponent, DropdownPosition } from './ng-select.component';
+import { ItemsList } from './items-list';
+
+declare var process: any;
+const IS_TEST = process.env.NODE_ENV === 'TEST';
+
+@Component({
+    providers: [VirtualScrollService],
```
Comment (varnastadeus) [inline src/ng-select/dropdown-panel.component.ts]: Here each component instance will have separate `VirtualScrollService` instance, not sure if it is good idea

**Comment by varnastadeus** [comment]
Looks nice, great job!

**Pull request merged**